File: Structural/Flyweight/CSharp/TextEditor.cs
// TextEditor.cs

using System;
using System.Collections.Generic;

public class TextEditor {
  private readonly List<FlyweightCharacter> characters = new List<FlyweightCharacter>();

  public void AddCharacter(char symbol) {
    characters.Add(CharacterFactory.GetCharacter(symbol));
  }

  public void DisplayCharacters() {
    foreach (FlyweightCharacter character in characters) {
      character.Display();
    }
    Console.WriteLine($"Total unique characters in memory: {CharacterFactory.GetPoolSize()}");
  }
}

File: Structural/Flyweight/JavaScript/textEditor.js
// textEditor.js

const CharacterFactory = require("./characterFactory");

class TextEditor {
  constructor() {
    this.characters = [];
  }

  addCharacter(symbol) {
    this.characters.push(CharacterFactory.getCharacter(symbol));
  }

  displayCharacters() {
    this.characters.forEach(character => character.display());
    console.log(`Total unique characters in memory: ${CharacterFactory.getPoolSize()}`);
  }
}

module.exports = TextEditor;
